streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -432,6 +437,10 @@ protected void maybeCommit() {
     private void commitAll() {
         for (StreamTask task : activeTasks.values()) {
             commitOne(task, time.milliseconds());
+            final ProcessorNode node = task.node();
```

Comment:
You are right, this was not the right place. StreamTask's commit is the right place. Fixed locally and working on a unit test and integration test to "count down" commit and punctuate metrics to ensure correctness.